src/components/App.js

```diff
@@ -246,6 +247,20 @@ class App extends Component {
             }
         })
 
+        ipcRenderer.on('updateLogger', (evt, ...args) => {
```

Comment:
The only place that sends this ipc event is in `PreferencesDrawer`. We can just execute this code there, can't we? That way, we don't need to make an ipc call.

Reply:
We can't. The logger instance is local to App.js, and since logger.js a module it isn't possible to access the logger instance in PreferencesDrawer, since that would need a remote.require(), which is only possible if logger.js wasn't a module.

Reply:
Why not? You can use `require` like all the other modules.

Reply:
Doing a normal require doesn't give me access to the logger.js instance that is created in App, which holds state information and the actual winston.logger instance.  We can only get away with doing a normal require as it is in PreferencesDrawer since we don't actually need to use any of that above: only the logger.js helper functions for validating the logging path are needed.

Reply:
> Let me put it another way, doing a normal require in PreferencesDrawer and trying to update without IPC would be the same as futilely trying to access the logger from a different process (they are different instances)

It's a common misconception that Node.js will load a module on each `require`. Actually Node.js [will cache the module object](https://nodejs.org/api/modules.html#modules_caching) and reuses it on each subsequent `require`.

If it puts your mind at ease, you can also access the "right" logger.js instance by `sabaki.modules.logger`.

Reply:
I see! Thanks for telling me that.

OK. I've changed it in the code.  This is resolved now with 08fbd0a